### PR TITLE
Conversion to package

### PR DESCRIPTION
**Pull request by avalentino**
On top of the changes implemented in #2 this PR converts `safecheck` into a `pip` installable package.

~Moreover a `mkzipapp.py` script is provided to generate a standalone executable that can be used without any installation (on unix systems).~

Closes #2

### CODE REVIEW

**Comment by svniemeijer** [comment]
Please remove the zipapp part of this merge request. I don't think this is useful.
Note that initially safecheck was using elementtree and the xmllint command tool, but with V3 it is using lxml, which now requires a special python environment to allow the `from lxml import` to work (also if you have the zipapp version). And because of this, it is better to ask users to install safecheck in the environment that also contains lxml.

**Comment by svniemeijer** [comment]
Other point, please don't use `from .safecheck import *` in `__init__.py`. Either just put the whole `safecheck.py` in `__init__.py` or import only those parts that comprise the public API when using safecheck as a library (something that is not officially supported/documented at the moment, but which we can introduce). In the latter case, `main` should not be part of this API. And the `entry_points` part should always point directly to the module that contains this `main` function (i.e. if `safecheck.py` stays a separate file then we should point to `safecheck.safecheck:main`).

**Comment by avalentino** [comment]
> Please remove the zipapp part of this merge request. I don't think this is useful.
> Note that initially safecheck was using elementtree and the xmllint command tool, but with V3 it is using lxml, which now requires a special python environment to allow the from lxml import to work (also if you have the zipapp version). And because of this, it is better to ask users to install safecheck in the environment that also contains lxml.

The rationale is that the zipapp executable has the same constraints/dependencies that the current version of the `safecheck.py` script has. Just it solves the problem of packing together the python code and the resource files (XSDs) that now are external.

But I agree with you that asking the users to install safecheck in a proper environment is the cleanest and safest solution.

I will remove the 'mkzipapp.py' script.

**Comment by svniemeijer** [comment]
The `__main__.py` script can then also be removed by the way.

**Comment by avalentino** [comment]
> The __main__.py script can then also be removed by the way.

It could be useful if you want to run the program as follows:

```
$ python3 -m safecheck [ARGS]
```

I use it quite a lot.

**Comment by svniemeijer** [comment]
I see one more issue. The `actual_crc = pathlib.Path(product).stem.split('_')[8]` part is incorrect, since it includes the path prefix. It should do a `split` on the `basename` without the path.

**Comment by svniemeijer** [comment]
Ah. No wait. That is not it. Let me check further

**Comment by svniemeijer** [comment]
Ok. So the problem is different. The 'split' assumes a fixed number of '_' occurrences, but this is not valid since a product type can have a dynamic number of `_` characters.

**Comment by svniemeijer** [comment]
I think the better approach would be to stick to using the old approach of using `stem[-4:]` but removing `_COG` from this stem if it was part of the product name.

File: safecheck/safecheck.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+def s1_check_product_crc(product, manifestfile):
+    """
+    Check CRC part of the product name.
+    Only applies to S1 products
+    """
+    expected_crc = format(binascii.crc_hqx(pathlib.Path(manifestfile).read_bytes(), 0xFFFF), '04X')
+    # Standard products the CRC is the last 4 chararcters in the productname, before the .SAFE extension
+    # This is as well the 9th subpart of product name while splitting using '_'
+    # On GRD/COG products, an additional '_COG' is added in product name
+    actual_crc = pathlib.Path(product).stem.split('_')[8]
```
Comment (avalentino) [inline safecheck/safecheck.py]:
```diff
-    actual_crc = pathlib.Path(product).stem.split('_')[8]
+    stem = pathlib.Path(product).stem
+    if stem.endswith("_COG"):
+        stem = stem[:-4]
+    actual_crc = stem[-4:]
```

@svniemeijer do you mean something like this?

Comment (svniemeijer) [reply safecheck/safecheck.py]: Yep. I think that will do it. I will test it anyway once it has been committed.

Comment (avalentino) [reply safecheck/safecheck.py]: done

**Pull request merged**